mlflow/tracking/metic_value_conversion_utils.py

```diff
@@ -0,0 +1,163 @@
+import sys
+
+from mlflow.exceptions import MlflowException
+
+
+def is_module_imported(module_name: str) -> bool:
+    return module_name in sys.modules
+
+
+def __try_get_item(x):
+    try:
+        xval = x.item()
+    except ValueError:
+        raise MlflowException(
+            f"Expected metric value to contain a single element, got shape {x.shape} instead"
+        )
+
+    return xval
+
+
+def __converter_requires(module_name: str):
+    """Wrapper function that checks if specified `module_name` is already imported before invoking wrapped function."""
+
+    def decorator(func):
+        def wrapper(x):
+            if not is_module_imported(module_name):
+                return x
+
+            return func(x)
+
+        return wrapper
+
+    return decorator
+
+
+def convert_metric_value_to_str_if_possible(x) -> str:
+    if x is None or type(x) == str:
+        return x
+
+    converter_fns_to_try = [
+        convert_metric_value_to_str_if_pyspark_mlparam,
+        convert_metric_value_to_str_if_ndarray,
+        convert_metric_value_to_str_if_tensorflow_tensor,
+        convert_metric_value_to_str_if_torch_tensor,
+    ]
+
+    for converter_fn in converter_fns_to_try:
+        possible_str = converter_fn(x)
+        if type(possible_str) == str:
+            return possible_str
+
+    return str(x)
+
+
+def convert_metric_value_to_float_if_possible(x, convert_int: bool = True) -> float:
```

Comment:
Looks like `convert_int = False` is never specified by any callers. Shall we remove it?

Looks like we always convert to float when logging metrics anyway:

```
import mlflow

from mlflow.tracking.client import MlflowClient

client = MlflowClient()

run = mlflow.start_run()
mlflow.log_metric("M", 1)

run_fetched = client.get_run(run.info.run_id)
print(run_fetched.data.metrics)
```

yields

```
1.0
```